File: POO JAVA/Aula06/src/aula06/ControleRemoto.java
package aula06;

public class ControleRemoto implements Controlador{

    private int volume;
    private boolean ligado;
    private boolean tocando;

    //Metodos especiais construtores
    public ControleRemoto() {
        this.volume = 50;
        this.ligado = true;
        this.tocando = false;
    }
    //Metodos especiais acessores
    private int getVolume() {
        return volume;
    }
    private void setVolume(int volume) {
        this.volume = volume;
    }
    private boolean getLigado() {
        return ligado;
    }
    private void setLigado(boolean ligado) {
        this.ligado = ligado;
    }
    private boolean getTocando() {
        return tocando;
    }
    private void setTocando(boolean tocando) {
        this.tocando = tocando;
    }
    
    //Metodos abstratos
    @Override
    public void ligar() {
        this.setLigado(true);
    }
    @Override
    public void desligar() {
        this.setLigado(false);
    }
    @Override
    public void abrirMenur() {
        System.out.println("------ Menu ------");
        System.out.println("Esta ligado? "+this.getLigado());
        System.out.println("Esta tocando? "+this.getTocando());
        System.out.print("Volume: "+this.getVolume());
        for(int i=0; i <= this.getVolume(); i +=10){
            System.out.print("[]");
        }
        System.out.println(" ");
    }
    @Override
    public void fecharMenur() {
        System.out.println("Fechando Menu ...");
    }
    @Override
    public void maisVolume() {
        if(this.getLigado()){
            this.setVolume(this.getVolume() + 5);
        } else{
            System.out.println("Impossivel aumentar volume.");
        }
    }
    @Override
    public void menosVolume() {
        if(this.getLigado()){
            this.setVolume(this.getVolume() - 5);
        } else{
            System.out.println("Impossivel abaixar volume.");
        }
    }
    @Override
    public void ligarMudo() {
        if(this.getLigado() && this.getVolume()>0){
            this.setVolume(0);
        }
    }
    @Override
    public void desligarMudo() {
        if(this.getLigado() && this.getVolume()==0){
            this.setVolume(55);
        }
    }
    @Override
    public void play() {
        if(this.getLigado() && !(this.getTocando())){
            this.setTocando(true);
        } else{
            System.out.println("Erro! Não consegui reproduzir.");
        }
    }
    @Override
    public void pause() {
        if(this.getLigado() && this.getTocando()){
            this.setTocando(false);
        } else{
            System.out.println("Erro! Não consegui pausar.");
        }
    }
}

File: POO PHP/Aula06/ControleRemoto.php
<?php
require_once './Controlador.php';

class ControleRemoto implements Controlador{

    private $volume;
    private $ligado;
    private $tocando;
    
    //Metodo construtor
    public function __construct() {
        $this->volume = 50;
        $this->ligado = false;
        $this->tocando = false;
    }
    //Metodos acessores
    private function getVolume() {
        return $this->volume;
    }
    private function getLigado() {
        return $this->ligado;
    }
    private function getTocando() {
        return $this->tocando;
    }

    private function setVolume($volume): void {
        $this->volume = $volume;
    }
    private function setLigado($ligado): void {
        $this->ligado = $ligado;
    }
    private function setTocando($tocando): void {
        $this->tocando = $tocando;
    }

    
    //Função da classe
    public function ligar() {
        $this->setLigado(true);
    }
    public function desligar() {
        $this->setLigado(false);
    }
    public function abrirMenur() {
        echo "<br>---------- Menu ----------";
        echo "<br>Está ligado? ".($this->getLigado()?"Sim.":"Não.");
        echo "<br>Está tocando? ".($this->getTocando()? "Sim.":"Não.");
        echo "<br>Volume? ".$this->getVolume();
        for($i=0; $i<= $this->getVolume(); $i+=10){
            echo "|";
        }
        echo "</br>";
    }
    public function fecharMenur() {
        echo "<br>Fechando Menu ...";
    }
    public function ligarMudo() {
        if($this->getLigado() && $this->getVolume()>0){
            $this->setVolume(0);
        }
    }
    public function desligarMudo() {
        if($this->getLigado() && $this->getVolume()==0){
            $this->setVolume(50);
        }
    }
    public function maisVolume() {
        if($this->getLigado()){
            $this->setVolume($this->getVolume() + 5);
        } else {
            echo "Erro. Aparecho desligado.";
        }
    }
    public function menosVolume() {
         if($this->getLigado()){
            $this->setVolume($this->getVolume() - 5);
        } else {
            echo "Erro. Aparelho desligado";
        }
    }
    public function pause() {
        if($this->getLigado() && $this->getTocando()){
            $this->setTocando(false);
        }
    }
    public function play() {
        if($this->getLigado() && !($this->getTocando())){
            $this->setTocando(true);
        }
    }
}